docs/proposals/generate-jks-certs-on-container-start.md

```diff
@@ -0,0 +1,80 @@
+# Generating JKS When Container Starts
+
+## Overview
+This proposal is the design to support [creating Java Keystores](https://trello.com/c/6a9bxQdX/) on container start from secrets that are mounted into the container.  This will:
+
+* Allow the logging stack to consume certificates provided by an OpenShift certificate generation service
+* Remove the dependency on Java for the OpenShift installer.
+* Remove the need for the OpenShift installer to generate certificates and Java Keystores
+* Allow the cluster to regenerate certificates when they expire
+
+## Assumptions
+* Something (e.g. intially OpenShift installer) will generate [client certificates](https://trello.com/c/QddRRfrS/#comment-58912b58b3005f71db0f81e6) and
+populate secrets.
+
+## Design
+### Elasticsearch Configuration
+Elasticsearch is configured such that it will utilize the keystore and truststore generated by the ```run.sh``` script at startup.
+
+### Container Resource Definition:
+Generated client certificates will be mounted as:
+```
+/etc/elasticsearch/secrets/client-cert/<ALIAS>/tls.{cert,key}
+```
+For example:
+```
+/etc/elasticsearch/secrets/client-cert
+  |- elasticsearch
```

Comment:
So would this be the mount point, and thus one secret for each client? Because I don't think you can get a directory hierarchy in a secret. If you don't want a secret per client then you may need to flatten this...